packages/connect/src/protocol-connect/handler-factory.spec.ts

```diff
@@ -51,51 +53,89 @@ describe("createHandlerFactory()", function () {
     },
   } as const;
 
-  function stub<M extends MethodInfo>(
-    opt: {
-      service?: ServiceType;
-      method?: M;
-      impl?: MethodImpl<M>;
-    } & Partial<UniversalHandlerOptions>
+  function setupTestHandler<M extends MethodInfo>(
```

Comment:
question: I'm wondering if you think there's anything here that's worth folding into `creatRouterTransport` instead?  seems fine like it is, but just would like to hear your thoughts on whether this is really a one-off that one we (in our codebases) would care about and not something a typical user would ever need.

Reply:
(after reading the rest of the PR, I see it's copied 3 times: if we don't upgrade createRouterTransport in some way to support the same use-cases, maybe it'd be good to extract to somewhere so the implementations don't drift?)

Reply:
Same code, but not identical, because it uses different imports. I only put the headers tests here because it was the least effort to get some coverage without updating our test infra. I think we're better off by testing the router transport the same way we test other transports (crosstests).